sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/eventlogger/Event.kt

```diff
@@ -19,32 +19,39 @@ internal data class Event(
     val rmcSdks: Map<String, String>? = null,
     val info: Map<String, String>? = null,
     var occurrenceCount: Int = 0,
-    @SerializedName("firstOccurrenceOn") var firstOccurrenceMillis: Long? = null
-) {
-    val eventVersion = "1.0"
-    val platform = Platform.ANDROID.displayName
-}
+    val eventVersion: String = "1.0",
+    val platform: String = "Android",
+    @SerializedName("firstOccurrenceOn") val createdOn: Long = System.currentTimeMillis()
+)
 
 // ---------------------------- Event enums ----------------------------
 internal enum class EventType(val displayName: String) {
     CRITICAL("0"),
     WARNING("1")
 }
 
-internal enum class Platform(val displayName: String) {
-    ANDROID("Android")
-}
-
 // ---------------------------- Event extensions ----------------------------
 /**
  * Returns the string hash based on some data from the [Event].
  */
-internal fun Event.getIdentifier() = "${this.appVersion}${this.sdkName}${this.errorCode}${this.errorMessage}"
+internal fun Event.generateEventIdentifier() = generateEventIdentifier(
+    eventType,
+    appVersion,
+    sdkName,
+    errorCode,
+    errorMessage
+)
+
+internal fun generateEventIdentifier(
+    eventType: String,
+    appVersion: String,
+    sdkName: String,
+    errorCode: String,
+    errorMessage: String
+) = "$eventType$appVersion$sdkName$errorCode$errorMessage"
     .getMD5HashData()
     .orEmpty()
```

Comment:
what is this `orEmpty()` supposed to do here ? 
if `getMD5HashData()` returns null => will id be empty there ? => if yes => can we have some other logic rather than returning empty ?

Reply:
May be current time in milliseconds appended by some other string => for the sake of uniquesness

Reply:
Yes it will be empty, but this is just for edge case and should not happen, as MD5 hash will always produce valid 32-character string

Reply:
I agree that it is most unlikely to happen that `getMD5HashData()` may returns `null` but still if we are checking for `null` -> then we should avoid returning an empty value -> as it may lead to set eventId as empty.

Reply:
So, I was thinking this => https://github.com/rakutentech/android-sdkutils/pull/67/files#r1414762847

Reply:
We cannot set current time in milliseconds since when we receive new similar event, it won't be able to compare the new event and the old event. For the sake of edge case processing, will just use the original `"$eventType$appVersion$sdkName$errorCode$errorMessage"` trimmed to 32- characters.

Reply:
Updated @puneetksh

Reply:
Okay, got it.